### ISSUE TITLE
ansible-lint doesn't recognize if a variable is only used in a block when it is defined

### ISSUE THREAD

**Issue by MarkusTeufelberger** [ansible-community/ansible-lint#506]
# Issue Type
- Bug report

# Ansible and Ansible Lint details

```
ansible --version
ansible 2.7.8

ansible-lint --version
ansible-lint 4.1.0
```

- ansible installation method: OS package
- ansible-lint installation method: pip

Reproducer:

```
- name: do something if foo is set
  when: foo is defined
  block:
    - name: write foo to disk
      copy:
        content: "{{ foo | to_nice_json }}\n"
        dest: /etc/foo/bar.json
```

# Desired Behaviour

Ansible-lint should just move on if the variable is not defined because the tasks in the block can't be executed in that case.

# Actual Behaviour (Bug report only)

Ansible-lint throws a warning:

` [WARNING]: Unable to convert data using to_nice_json, falling back to to_json: Undefined is not JSON serializable`

(It would be very useful by the way to know the file and line or at least the variable name that causes this warning!)

**Comment by arxcruz** [comment]
I'm also having this problem and the workaround (at least for me) is to add a default() like "{{ foo | default('') | to_nice_json }}"

**Comment by ssbarnea** [comment]
I am unable to reproduce this error with v5.0.0
